src/hangar/checkout.py

```diff
@@ -127,6 +134,236 @@ def __verify_checkout_alive(self):
+
+        Get samples from all arraysets (shortcut syntax)
+
+            >>> out = dset[:, ('1', '2')]
+            >>> out = dset[..., ('1', '2')]
+            >>> out
+            [ArraysetData(foo=array([1]), bar=array([11]), baz=array([111])),
+             ArraysetData(foo=array([2]), bar=array([22]), baz=array([222]))]
+
+            >>> out = dset[:, '1']
+            >>> out = dset[..., '1']
+            >>> out
+            ArraysetData(foo=array([1]), bar=array([11]), baz=array([111]))
+
+        Parameters
+        ----------
+        index
+            Please see detailed explanation above for full options. Hard coded
+            options are the order to specification.
+
+            The first element (or collection) specified must be ``str`` type and
+            correspond to an arrayset name(s). Alternativly the Ellipsis operator
+            (``...``) or unbounded slice operator (``:`` <==> ``slice(None)``) can
+            be used to indicate "select all" behavior.
+
+            If a second element (or collection) is present, the keys correspond to
+            sample names present within (all) the specified arraysets. If a key is
+            not present in even on arrayset, the entire ``get`` operation will
+            abort with ``KeyError``. If desired, the same selection syntax can be
+            used with the :meth:`~hangar.checkout.ReaderCheckout.get` method, which
+            will not Error in these situations, but simply return ``None`` values
+            in the appropriate position for keys which do not exist.
+
+        Returns
+        -------
+        Arrayset
+            single arrayset parameter, no samples specified
+
+        np.ndarray
+            Single arrayset specified, single sample key specified
+
+        List[np.ndarray]
+            Single arrayset, multiple samples array data for each sample is
+            returned in same order sample keys are recieved.
+
+        List[NamedTuple[*np.ndarray]]
+            Multiple arraysets, multiple samples. Each arrayset's name is used
+            as a field in the NamedTuple elements, each NamedTuple contains
+            arrays stored in each arrayset via a common sample key. Each sample
+            key is returned values as an individual element in the
+            List. The sample order is returned in the same order it wasw recieved.
+
+
+        Notes
+        -----
+
+        *  All specified arraysets must exist
+
+        *  All specified sample `keys` must exist in all specified arraysets,
+           otherwise standard exception thrown
+
+        *  Slice syntax cannot be used in sample `keys` field
+
+        *  Slice syntax for arrayset field cannot specify `start`, `stop`, or
+           `step` fields, it is soley a shortcut syntax for 'get all arraysets' in
+           the ``:`` or ``slice(None)`` form
+
+        .. seealso:
+
+            :meth:`~hangar.checkout.ReaderCheckout.get`
+
+        """
+        try:
```

Comment:
IMO, we should offload all of these tasks to `get` method and `__getitem__` will just do the `index` parsing and call `get`. This way we can avoid double checking and make sure our semantics never is broken.